JASP-Engine/JASP/R/reliabilityanalysis.R

```diff
@@ -15,479 +15,456 @@
-
-				message <- paste(message, "Warning: Greatest lower bound can only be calculated for three or more variables.")
-
-			} else if (isTryError(r[["glb"]])) {
-
-				message <- paste(message, "Warning: Greatest lower bound could not be calculated.")
-
-			}
-
-		}
-
-		.addFootnote(footnotes, symbol = "<em>Note.</em>", text = message)
-
-		table[["footnotes"]] <- as.list(footnotes)
-
-
-		alpha <- NULL
-		lambda <- NULL
-		mu <- NULL
-		sd <- NULL
-		rho <- NULL
-		omega <- NULL
-		glb <- NULL
-		lower <- NULL
-		upper <- NULL
-
-		if (options[["alphaScale"]])
-            alpha <- .clean(r$total[[options$alphaNms]])
-
-		if (options$gutmannScale)
-			lambda <- .clean(r$total[["G6(smc)"]])
-
-		if (options$meanScale)
-			mu <- .clean(r$total$mean)
-
-		if (options$sdScale)
-			sd <- .clean(r$total$sd)
-
-		if (options[["averageInterItemCor"]])
-			rho <- .clean(r[["total"]][["average_r"]])
-
-		if (options[["mcDonaldScale"]])
-			omega <- .clean(r[["omega"]])
-
-		if (options[["glbScale"]]) {
-			if (r[["glb"]] == "." || isTryError(r[["glb"]])) { # unusable information
-				glb <- "."
-			} else { # a useable value
-				glb <- .clean(r[["glb"]])
-			}
-		}
-
-		if (options[["confAlpha"]]) {
-			lower = .clean(r[["ciAlpha"]][1])
-			upper = .clean(r[["ciAlpha"]][2])
-		}
-
-		data[[1]] <- list(case="scale", alpha=alpha, lambda=lambda, omega = omega, glb = glb, rho=rho, mu=mu, sd=sd, lower = lower, upper = upper)
-
-		table[["status"]] <- "complete"
-
-	} else {
-
-		data[[1]] <- list(case="scale", alpha=".", lambda=".", omega = ".", glb = ".", rho =".", mean=".", sd=".", lower = ".", upper = ".")
-
-	}
-
-	table[["data"]] <- data
-
-	return(table)
-
+.reliabilityScaleFill <- function(jaspResults, dataset, options, relyFit, ready) {
+  results   <- list()
+  nObs      <- nrow(dataset)
+  nVar      <- ncol(dataset)
+  variables <- unlist(options$variables)
+  options   <- .reliabilitySetAlphaNms(options)
+  
+  if(ready){
+    #Initialise all as null
+    alpha <- lambda <- omega <- glb <- rho <- mu <- sd <- lower <- upper <- NULL
+    
+    if(nVar >= 2){
+      if (options$alphaScale)
+        alpha  <- relyFit$total[[options$alphaNms]]
+      if (options$gutmannScale)
+        lambda <- relyFit$total[["G6(smc)"]]
+      if (options$meanScale)
+        mu     <- relyFit$total$mean
+      if (options$sdScale)
+        sd     <- relyFit$total$sd
+      if (options$averageInterItemCor)
+        rho    <- relyFit$total$average_r
+      if (options$mcDonaldScale)
+        omega  <- relyFit$omega
+      if (options$glbScale) {
+        if (relyFit$glb == "." || isTryError(relyFit$glb)) # unusable information
+          glb  <- "."
+        else # a useable value
+          glb  <- relyFit$glb
+      }
+      if (options[["confAlpha"]]) {
+        lower <- relyFit[["ciAlpha"]][1]
+        upper <- relyFit[["ciAlpha"]][2]
+      }
+    }
+  } else #Initialise all as "."
+    alpha <- lambda <- omega <- glb <- rho <- mu <- sd <- lower <- upper <- "."
```

Comment:
This is more of an old-style kinda thing. Returning a table without any rows generates one row of dots for you automatically.